10_rest/n0npax/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	puppy "github.com/anz-bank/go-course/10_rest/n0npax/pkg/puppy"
+	store "github.com/anz-bank/go-course/10_rest/n0npax/pkg/puppy/store"
+	"github.com/gin-gonic/gin"
+)
+
+var (
+	out io.Writer = os.Stderr
+)
+
+func main() {
+	config, err := parseArgs()
+	puppyFilePath, storeType, port := config.path, config.sType, config.port
+	exitOnError(err)
+
+	storer, err := createStorer(storeType)
+	exitOnError(err)
+
+	if puppyFilePath != "" {
+		file, err := os.Open(puppyFilePath)
+		exitOnError(err)
+		puppies, err := readPuppies(file)
+		exitOnError(err)
+		err = feedStore(storer, puppies)
+		exitOnError(err)
+	}
+	r := puppy.RestBackend(storer)
+	err = runSrv(r, port)
+	exitOnError(err)
```

Comment:
I think I'd factor this whole thing differently:

```go
func main() {
    log.Fatal(runPuppyServer(Os.Args[1:]))
}

func servePuppyServer(args []string) error {
    config, err = parseArgs(args)
    if err != nil { 
         return err
    }
    store, err := createStore(config.storeType)
    if err != nil { 
         return err
    }
    if err := feedStoreFromFile(config.PuppyFile); err != nil {
        return err
     }
     return puppy.RestBackend(store).Run(" <message> ", port)
}
```

Reply:
fixed. Implemented very similar approach